src/utils/useAtomsSnapshot.ts

```diff
@@ -115,7 +121,11 @@ export function useAtomsSnapshot({
       }
       prevValues = values;
       prevDependents = dependents;
-      setAtomsSnapshot({ values, dependents });
+      const deferrableAtomSetAction = () =>
+        setAtomsSnapshot({ values, dependents });
+      deferAtomSetActions
+        ? () => Promise.resolve().then(deferrableAtomSetAction)
```

Comment:
```suggestion
        ? Promise.resolve().then(deferrableAtomSetAction)
```

btw, I don't know if we prefer the ternary expression.

Reply:
Fixed, waiting for opinions over ternary operator

Reply:
@arjunvegda

Reply:
Yes, would it be possible to switch to `if/else` that'd be a little more readable.

Reply:
Also, on second thought, this feels like a hack 🤔 @dai-shi wouldn't our consumers for `jotai` run into the same issue? Do you think we should fix this in Jotai?

Reply:
This is an edge-case issue only with the `dev_*` methods, which devtools only suffers from.
I originally thought about fixing it in`jotai`, but that's somewhat limiting the capability. So, I think it's better to fix it here.

To make it less hacky, we could remove `useRef` and always delay a microtask. what do you think? @yf-yang

Reply:
At least, moving everything to the next microtask would break current tests.

If hacky means readability issue, I think I can use clearer naming and comments to ease the problem.

Reply:
> At least, moving everything to the next microtask would break current tests.

Oh, I see.